src/routes/products/functions/(components)/RegionsMap.svelte

```diff
@@ -0,0 +1,286 @@
+        </p>
+    </div>
+    <div
+        class="container relative mx-auto flex h-full items-center justify-center py-10"
+        bind:this={mapContainer}
+    >
+        {#if showTooltip}
+            <div
+                bind:this={tooltipElement}
+                in:fly={{ y: 10, duration: 500 }}
+                out:fly={{ y: -10, duration: 500 }}
+                class="pointer-events-none absolute z-100 block flex w-[190px] flex-col gap-2 rounded-[10px] border border-white bg-gradient-to-br from-white/64 to-white/32 p-2 backdrop-blur-sm"
+                style:left="{tooltip.x}px"
+                style:top="{tooltip.y - (tooltipElement?.offsetHeight || 0) - 10}px"
+            >
+                <span class="text-primary text-caption w-fit"
+                    >{tooltip.city}
+                    ({tooltip.code})</span
+                >
+                {#if tooltip.available}
+                    <div
+                        class="text-caption flex h-5 items-center justify-center place-self-start rounded-[6px] bg-[#10B981]/16 p-1 text-center text-[#0A714F]"
+                    >
+                        <span class="text-micro -tracking-tight">Available now</span>
+                    </div>
+                {:else}
+                    <div
+                        class="text-caption flex h-5 items-center justify-center place-self-start rounded-[6px] bg-black/6 p-1 text-center text-[#56565C]"
+                    >
+                        <span class="text-micro -tracking-tight">{tooltip.release}</span>
+                    </div>
+                {/if}
+            </div>
+        {/if}
+        <div class="relative z-10 block w-full space-y-4 md:hidden">
+            {#each pins as pin}
+                <div
+                    class="z-100 block flex w-full flex-col gap-2 rounded-[10px] border border-white bg-gradient-to-br from-white/64 to-white/32 p-3 backdrop-blur-sm"
+                >
+                    <span class="text-primary text-caption w-fit"
+                        >{pin.city}
+                        ({pin.code})</span
+                    >
+                    {#if pin.available}
+                        <div
+                            class="text-caption flex h-5 items-center justify-center place-self-start rounded-[6px] bg-[#10B981]/16 p-1 text-center text-[#0A714F]"
+                        >
+                            <span class="text-micro -tracking-tight">Available now</span>
+                        </div>
+                    {:else}
+                        <div
+                            class="text-caption flex h-5 items-center justify-center place-self-start rounded-[6px] bg-black/6 p-1 text-center text-[#56565C]"
+                        >
+                            <span class="text-micro -tracking-tight">{pin.release}</span>
+                        </div>
+                    {/if}
+                </div>
+            {/each}
+        </div>
+        <svg
```

Comment:
This is quite a big SVG to put in the file I think, can we move it to a separate file?

Reply:
I can, it just feels like it's easier to encapsulate it to this component rather than dealing with prop drilling. Thoughts?